pykrige/ok.py

```diff
@@ -288,29 +302,65 @@ def update_variogram_model(self, variogram_model, variogram_parameters=None,
 
         Parameters
         __________
-        variogram_model (string): May be any of the variogram models listed
+        variogram_model : str or GSTools CovModel
+            May be any of the variogram models listed
             above. May also be 'custom', in which case variogram_parameters
             and variogram_function must be specified.
-        variogram_parameters (list or dict, optional): List or dict of
+            You can also use a GSTools
+            CovModel, see https://github.com/GeoStat-Framework/GSTools
+        variogram_parameters : list or dict, optional
+            List or dict of
             variogram model parameters, as explained above. If not provided,
             a best fit model will be calculated as described above.
-        variogram_function (callable, optional): A callable function that must
+        variogram_function : callable, optional
+            A callable function that must
             be provided if variogram_model is specified as 'custom'.
             See above for more information.
-        nlags (int, optional): Number of averaging bins for the semivariogram.
+        nlags : int, optional
+            Number of averaging bins for the semivariogram.
             Default is 6.
-        weight (boolean, optional): Flag that specifies if semivariance at
+        weight : boolean, optional
+            Flag that specifies if semivariance at
             smaller lags should be weighted more heavily when automatically
             calculating the variogram model. See above for more information.
             True indicates that weights will be applied. Default is False.
-        anisotropy_scaling (float, optional): Scalar stretching value to
+        anisotropy_scaling : float, optional
+            Scalar stretching value to
             take into account anisotropy. Default is 1 (effectively no
             stretching). Scaling is applied in the y-direction.
-        anisotropy_angle (float, optional): CCW angle (in degrees) by
+        anisotropy_angle : float, optional
+            CCW angle (in degrees) by
             which to rotate coordinate system in order to take into
             account anisotropy. Default is 0 (no rotation).
         """
 
+        # set up variogram model and parameters...
+        self.variogram_model = variogram_model

```

Comment:
Can we factorize this code repeated in all estimators, maybe as,
```py
def _setup_variogram_parameters(self, ..)
```
under pykrige.core. There is already an `_initialize_variogram_model` there. All that could probably be made methods of some base class, but that should probably be done in a separate PR..

Reply:
I just tried that, but it turns out to be more difficult than thought. It should be done in a separate PR refactoring the kriging classes. The ``update_variogram_model`` method should be used here, but it has to be adopted, to work with different dimensions. I will leave it as it is now, if that is ok.